spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt

```diff
@@ -0,0 +1,150 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.adevinta.spark.components.progress
+
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ProgressIndicatorDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import com.adevinta.spark.InternalSparkApi
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.components.progress.ProgressIndicatorDefaults.CircularSize
+import com.adevinta.spark.tools.modifiers.sparkUsageOverlay
+import com.adevinta.spark.tools.preview.SparkPreviewProvider
+import com.adevinta.spark.tools.preview.ThemeVariant
+import com.adevinta.spark.tools.preview.UserType
+import androidx.compose.material3.CircularProgressIndicator as MaterialCircularProgressIndicator
+
+@InternalSparkApi
+@Composable
+fun SparkCircularProgressIndicator(
+    progress: Float,
+    isIndeterminate: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    val color = ProgressIndicatorDefaults.circularColor // Primary
+    if (isIndeterminate) {
+        MaterialCircularProgressIndicator(
+            modifier = modifier
+                .sparkUsageOverlay()
+                .size(CircularSize),
+            color = color,
+        )
+    } else {
+        MaterialCircularProgressIndicator(
+            modifier = modifier
+                .sparkUsageOverlay()
+                .size(CircularSize),
+            progress = progress,
+            color = color,
+        )
+    }
+}
+
+/**
+ * <a href="https://github.mpi-internal.com/leboncoin/android-app/blob/delivery/libraries/Compose/ProgressIndicators.md" class="external" target="_blank">Determinate Spark linear progress indicator</a>.
+ *
+ * Progress indicators express an unspecified wait time or display the duration of a process.
+ *
+ * By default there is no animation between [progress] values. You can use
+ * [ProgressIndicatorDefaults.ProgressAnimationSpec] as the default recommended [AnimationSpec] when
+ * animating progress
+ *
+ * @param progress the progress of this progress indicator, where 0.0 represents no progress and 1.0
+ * represents full progress. Values outside of this range are coerced into the range.
+ * @param modifier the [Modifier] to be applied to this progress indicator
+ */
+@Composable
+fun CircularProgressIndicator(
+    progress: Float,
```

Comment:
I think we should provide a FloatRange to help devs catch a potential error as soon as possible.

Reply:
Yep I agree 👍